storm-client/src/jvm/org/apache/storm/daemon/worker/BackPressureTracker.java

```diff
@@ -118,8 +118,26 @@ public void setLastOverflowCount(BackpressureState state, int value) {
         private int lastOverflowCount = 0;
 
 
-        BackpressureState(JCQueue queue) {
+        BackpressureState(JCQueue queue, Integer taskId, String componentId, StormMetricRegistry metricRegistry) {
             this.queue = queue;
+
+            // System bolt is not a part of backpressure.
+            if (taskId >= 0) {
+                if (componentId == null) {
+                    throw new RuntimeException("Missing componentId for task " + taskId);
+                }
+
+                Gauge<Integer> bpOverflowCount = new Gauge<Integer>() {
+                    @Override
+                    public Integer getValue() {
+                        if (backpressure.get()) {
+                            return Math.max(1, lastOverflowCount);
+                        }
+                        return 0;
+                    }
+                };
+                metricRegistry.gauge("__backpressure-overflow-count", bpOverflowCount, componentId, taskId);
```

Comment:
Can we change this to `__backpressure-last-overflow-count`? I think it reflects better the meaning of the metric. And we have a queue `-overflow` metric. Thinking the name change can make it a little bit more clear.

Reply:
ok

Reply:
updated

Reply:
Thanks